master/id_allocator.go

```diff
@@ -216,21 +203,23 @@ errHandler:
 func (alloc *IDAllocator) allocateClientID() (clientID uint64, err error) {
 	alloc.mpIDLock.Lock()
 	defer alloc.mpIDLock.Unlock()
-	var cmd []byte
-	metadata := new(RaftCmd)
-	metadata.Op = opSyncAllocClientID
-	metadata.K = maxClientIDKey
-	clientID = atomic.LoadUint64(&alloc.clientID) + 1
-	value := strconv.FormatUint(uint64(clientID), 10)
-	metadata.V = []byte(value)
-	cmd, err = metadata.Marshal()
-	if err != nil {
+	var leader uint64
+	var term uint64
+	var counterPart uint32
+	leader, term = alloc.partition.LeaderTerm()
+	if leader != alloc.partition.Status().NodeID {
 		goto errHandler
 	}
-	if _, err = alloc.partition.Submit(cmd); err != nil {
+	counterPart = alloc.clientCounter
+	if counterPart == math.MaxUint32 {
+		// FIXME: give up current term
+		err = ErrClientIdRunOut
 		goto errHandler
 	}
-	alloc.setClientID(clientID)
+	counterPart += 1
+	clientID = term << 32
+	clientID |= uint64(counterPart)
+	alloc.clientCounter = counterPart
 	return
```

Comment:
here, you can use `alloc.clientCounter ++`, no need to use a tmp variable counterPart

Reply:
Yes, I will fix it